packages/block-library/src/navigation/index.php

```diff
@@ -132,6 +132,84 @@ function block_core_navigation_render_submenu_icon() {
 	return '<svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" role="img" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg>';
 }
 
+
+/**
+ * Finds the first non-empty `wp_navigation` Post.
+ *
+ * @return WP_Post|null the first non-empty Navigation or null.
+ */
+function block_core_navigation_get_non_empty_navigation() {
+	$navigation_posts = get_posts(
+		array(
+			'post_type' => 'wp_navigation',
+		)
+	);
+
+	$navigation_post = null;
+
+	// Pick first non-empty Navigation.
+	foreach ( $navigation_posts as $navigation_maybe ) {
+		if ( ! empty( $navigation_maybe->post_content ) ) {
+			$navigation_post = $navigation_maybe;
+			break;
+		}
+	}
+
+	return $navigation_post;
+}
+
+/**
+ * 'parse_blocks' includes a null block with '\n\n' as the content when
+ * it encounters whitespace. This code strips it.
+ *
+ * @param array $parsed_blocks the parsed blocks to be normalized.
+ * @return array the normalized parsed blocks.
+ */
+function block_core_navigation_normalize_parsed_blocks( $parsed_blocks ) {
+	$filtered = array_filter(
```

Comment:
I understand the comment on what the function does, but when is this happening? Is it a bug? Is it a workaround?

Reply:
No this is just how the parser is supposed to work. I will add a comment to clarify.

Reply:
Added https://github.com/WordPress/gutenberg/pull/36740/commits/0ca7eb188660bb46253e2dd9844e94ca189235f4